src/content/posts/2021/pitfalls-of-unoptimzied-nuget-packages/index.md

```diff
@@ -0,0 +1,177 @@
+<br>
+<div align="center">
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">StructureMap 4.4.1 is up. No code changes, but corrects the problem of the nuget being built as Debug for 4.3.0 and 4.4.0. Oops on my part.</p>&mdash; jeremydmiller (@jeremydmiller) <a href="https://twitter.com/jeremydmiller/status/777571510919630848?ref_src=twsrc%5Etfw">September 18, 2016</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">I knew this would happen. I&#39;ve done it myself and you won&#39;t be the last. More debug builds on NuGet ahead. <a href="https://twitter.com/davidfowl?ref_src=twsrc%5Etfw">@davidfowl</a></p>&mdash; Kristian Hellang (@khellang) <a href="https://twitter.com/khellang/status/777753550491353088?ref_src=twsrc%5Etfw">September 19, 2016</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+</div>
+<br>
+{{< /rawhtml >}}
+
+## Anything that can go wrong will go wrong - [Kristian Murphy Hellang](https://twitter.com/khellang)
+
+I feel like publishing packages to _nuget.org_ relates a lot to sailing and running aground:
+
+> There are three kinds of skippers, those who have run aground, those who will run aground, and those that have but won't admit it.
+
+Replace _skippers_ with _package authors_ and _running aground_ with _publish packages built in debug_ and there you have it.
+
+Publishing unoptimized bits also happened to Sentry's _SharpRaven_ SDK back in 2018. If you're using [SharpRaven version 2.3.0](https://www.nuget.org/packages/SharpRaven/2.3.0), you're running an unoptimized assembly. I would recommend you [upgrade it to 2.3.1](https://github.com/getsentry/raven-csharp/pull/226/files) but on that same year that package was replaced by [Sentry](https://www.nuget.org/packages/Sentry/). Don't worry, it's [packaged in release mode](https://github.com/getsentry/sentry-dotnet/blob/bc4c5e23eaaaa848a9a44ae05ac257a591624b24/build.ps1#L14). Not that you should take my word for it.
+
+In early 2020, I even [opened a PR to add this verification to the Sentry SDK for .NET](https://github.com/getsentry/sentry-dotnet/pull/365/files) so it would check assemblies being loaded whether they were optimized or not, and send events to Sentry when applicable. But it never felt like the right thing to do. This verification must happen much earlier, at latest during build time.
+
+## Release build and compiler optimization
+
+I've been talking about packaging NuGet with `Release` configuration then talking about Roslyn optimization.
+These two things are not exactly the same (though by default related) and I wish to clarify.
+
+The `Release` configuration, by default does provide the `-optimize+` flag to the C# compiler 
+but you can achieve the same for example by means of a MSBuild property `<Optimize>true</Optimize>`.
+I hope the references to the docs above already made this clear.
```

Comment:
No, nobody reads the references unless they care TOO much ;)

Reply:
I mean the two snippets from the docs that I pasted. Talks about exactly this point

Reply:
I'll reword

Reply:
> I hope the quotes from the docs I included above helped make this clear.